nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -80,31 +84,70 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self.cfg = cfg
 
+        frozen_model_cfg = MegatronGPTModel.restore_from(
+            cfg.get('language_model_path'), trainer=trainer, return_config=True
+        )
+
+        # Need to overwrite some params in frozen model's config before restoring
+        with open_dict(frozen_model_cfg):
+            frozen_model_cfg.micro_batch_size = self.cfg.micro_batch_size
+            frozen_model_cfg.global_batch_size = self.cfg.global_batch_size
+            frozen_model_cfg.precision = trainer.precision
+
         # Load pretrained GPT model and tokenizer
         if cfg.get('language_model_path', None):
             self.frozen_model = MegatronGPTModel.restore_from(
-                cfg.get('language_model_path'), trainer=trainer, save_restore_connector=NLPSaveRestoreConnector(),
+                cfg.get('language_model_path'),
+                trainer=trainer,
+                save_restore_connector=NLPSaveRestoreConnector(),
+                override_config_path=frozen_model_cfg,
             )
 
+        self.float_type = torch.float
+
+        # Make prompt learning model able to load gpt models trained with amp_o2
+        if self.frozen_model.megatron_amp_o2:
+            self.frozen_model.model = self.frozen_model.model.module
+
+            if self.frozen_model.cfg.precision == 16:
+                self.float_type = torch.float16
+            elif self.frozen_model.cfg.precision == 'bf16':
+                self.float_type = torch.bfloat16
+            else:
+                raise ValueError(f'Precision {self.frozen_model.precision} Not supported with O2')
+
         # Freeze all GPT model weights for prompt-tuning/p-tuning
         self.frozen_model.freeze()
```

Comment:
We don't need to freeze the model if we're setting the LR to 0 for it's weights?

Reply:
Removed it